kroma-validator/l2_output_submitter.go

```diff
@@ -268,30 +340,65 @@ func (l *L2OutputSubmitter) CalculateWaitTime(ctx context.Context, nextBlockNumb
 	return 0
 }
 
-// HasEnoughDeposit checks if validator has enough deposit to bond when trying output submission.
-func (l *L2OutputSubmitter) HasEnoughDeposit(ctx context.Context) (bool, error) {
+// CanSubmitOutput checks if the validator can submit L2Output.
+func (l *L2OutputSubmitter) CanSubmitOutput(ctx context.Context, outputIndex *big.Int) (bool, error) {
 	cCtx, cCancel := context.WithTimeout(ctx, l.cfg.NetworkTimeout)
 	defer cCancel()
 	from := l.cfg.TxManager.From()
-	balance, err := l.valpoolContract.BalanceOf(optsutils.NewSimpleCallOpts(cCtx), from)
-	if err != nil {
-		return false, fmt.Errorf("failed to fetch deposit amount: %w", err)
-	}
-	l.metr.RecordDepositAmount(balance)
+	if l.poolTerminationIndex.Cmp(outputIndex) < 0 {
```

Comment:
I think it would be nice to make this a function.

```go
func (l *L2OutputSubmitter) IsValPoolTerminated(outputIndex *big.Int) {
    return l.poolTerminationIndex.Cmp(outputIndex) < 0
}
```